partcad/src/partcad/converters/stl_converter.py

```
@@ -0,0 +1,100 @@
+
+class StlConverter(BaseConverter):
+    def read(self) -> TopoDS_Shape:
+        """Reads an STL file and returns a TopoDS_Shape."""
+        stl_reader = StlAPI_Reader()
+        shape = TopoDS_Shape()
+        if not stl_reader.Read(shape, str(self.input_file)):
+            raise RuntimeError(f"Failed to read STL file: {self.input_file}")
+        return shape
+
+
+    def write(self, shape: TopoDS_Shape, tolerance=0.5, angular_tolerance=0.5, ascii=False) -> None:
```

Ruff (0.8.2)

Missing type annotation for function argument `tolerance`

(ANN001)